addons/project/models/project.py

```diff
@@ -494,11 +494,12 @@ def _compute_milestone_reached_count(self):
         for project in self:
             project.milestone_count_reached = mapped_count.get(project.id, 0)
 
-    @api.depends('milestone_ids', 'milestone_ids.is_reached', 'milestone_ids.deadline')
+    @api.depends('milestone_ids', 'milestone_ids.is_reached', 'milestone_ids.deadline', 'allow_milestones')
     def _compute_is_milestone_exceeded(self):
         today = fields.Date.context_today(self)
         read_group = self.env['project.milestone']._read_group([
             ('project_id', 'in', self.ids),
+            ('project_id.allow_milestones', '=', True),
```

Comment:
I think when we will have many milestones in the db, the query will take more time.
we could directly filter the project in self to only have the ones with `allow_miletones=True`

```suggestion
            ('project_id', 'in', self.filtered('allow_milestones').ids),
```

Reply:
Can you also reword your commit message?

```txt
[FIX] project: filter projects for `Late Milestones` filter


Before this commit, when the user uses the `Late Milestones` 
filter in the project views, all the projects in the views 
are used, but some projects could have the feature disabled
and so those ones should be ignored in the filtered.

This commit filters the projects to only apply the filter 
on those projects and not also the ones without the feature 
enabled.

task-3033041
```

Thanks in advance! 🙂